cts-java/src/test/java/org/ga4gh/cts/api/reads/ReadMethodsEndpointAliveIT.java

```diff
@@ -0,0 +1,179 @@
+import org.ga4gh.methods.SearchReadsResponse;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.net.HttpURLConnection;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.ga4gh.cts.api.Utils.aSingle;
+import static org.ga4gh.cts.api.Utils.catchGAWrapperException;
+
+/**
+ * <p>Verifies basic sanity of the reads/search API.</p>
+ * <p>The {@code READS} API methods (as defined by the {@code readmethods.avdl}) are:</p>
+ * <ul>
+ * <li>POST reads/search of SearchReadsRequest yields SearchReadsResponse</li>
+ * <li>POST /readgroupsets/search of SearchReadGroupSetsRequest yields SearchReadGroupSetsResponse</li>
+ * </ul>
+ *
+ * <p>The test invokes a search request with null, default, and error parameters
+ * on the endpoint and verifies the response. For tests with more insight into
+ * the data returned (complex queries, etc) refer to the ReadsSearchIT tests.</p>
+ *
+ * <p>The test shows simply making an assertion about the existence of the return itself
+ * (in {@code defaultReadgroupsetsRequestGetsResponse} and shows making an assertion about
+ * a field of the return (in {@code defaultReadsRequestGetsNullAlignments})</p>
+ *
+ * <p>Created by Wayne Stidolph on 5/20/15.</p>
+ */
+@Category(ReadsTests.class)
+@RunWith(JUnitParamsRunner.class)
+public class ReadMethodsEndpointAliveIT implements CtkLogs {
+
+    private static Client client = new Client(URLMAPPING.getInstance());
+
+    /**
+     * Show that a SearchReadsRequest is accepted and
+     * returns a parseable Response.
+     *
+     * @throws Exception the exception
+     */
+    @Test
+    public void defaultReadsRequestGetsNullAlignments() throws Exception {
+
+        // first get a valid reference to map our read to
+        final String refId = Utils.getValidReferenceId(client);
+
+        final String readGroupId = Utils.getReadGroupId(client);
+
+        // then do a read search
+        final SearchReadsRequest srr =
+                SearchReadsRequest.newBuilder()
+                                  .setReadGroupIds(aSingle(readGroupId))
+                                  .setStart(0L)
+                                  .setEnd(150L)
+                                  .setReferenceId(refId)
+                                  .build();
+        final SearchReadsResponse rtn = client.reads.searchReads(srr);
+        assertThat(rtn.getAlignments()).isNullOrEmpty();
+    }
+
+    /**
+     * Check that an unmatched {@link SearchReadsRequest} results in a thrown exception.
+     *
+     * @param readGroupId the ReadGroup ID, passed in using the @Parameters annotation
+     * @throws Exception the exception
+     */
+    @SuppressWarnings("ThrowableResultOfMethodCallIgnored")
+    @Test
+    @Parameters({
+            "myNonsenseId", "foo", ""
```

Comment:
Please file a ticket on making the spec clear and add the Documentation label.


CH Albach <notifications@github.com> writes:
> In cts-java/src/test/java/org/ga4gh/cts/api/reads/
> ReadMethodsEndpointAliveIT.java:
> 
> > +                                  .setReferenceId(refId)
> > +                                  .build();
> > +        final SearchReadsResponse rtn = client.reads.searchReads(srr);
> > +        assertThat(rtn.getAlignments()).isNullOrEmpty();
> > +    }
> > +
> > +    /**
> > +     * Check that an unmatched {@link SearchReadsRequest} results in a thrown exception.
> > +     *
> > +     * @param readGroupId the ReadGroup ID, passed in using the @Parameters annotation
> > +     * @throws Exception the exception
> > +     */
> > +    @SuppressWarnings("ThrowableResultOfMethodCallIgnored")
> > +    @Test
> > +    @Parameters({
> > +            "myNonsenseId", "foo", ""
> 
> I say we remove the third case. If we make the spec stricter on this point, we
> can add corresponding compliance test coverage.
> 
> —
> Reply to this email directly or view it on GitHub.*
>